File: src/main/java/com/und/security/RestUserFactory.java
package com.und.security;

import java.util.List;
import java.util.stream.Collectors;

import com.und.model.security.Authority;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.authority.SimpleGrantedAuthority;
import com.und.model.security.User;

public final class RestUserFactory {

    private RestUserFactory() {
    }

    public static EventUser create(User user) {
        return new EventUser(
                user.getId(),
                user.getUsername(),
                user.getFirstname(),
                user.getLastname(),
                user.getEmail(),
                user.getPassword(),
                mapToGrantedAuthorities(user.getAuthorities()),
                user.getEnabled(),
                user.getLastPasswordResetDate(),
                user.getClientSecret(),
                user.getKey()
        );
    }

    private static List<GrantedAuthority> mapToGrantedAuthorities(List<Authority> authorities) {
        return authorities.stream()
                .map(authority -> new SimpleGrantedAuthority(authority.getName().name()))
                .collect(Collectors.toList());
    }
}

File: src/main/kotlin/com/und/security/RestUserFactory.kt
package com.und.security


import com.und.security.model.Authority
import com.und.security.model.User
import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.authority.SimpleGrantedAuthority

class RestUserFactory {
    companion object {

        @JvmStatic
        fun create(user: User): UndUserDetails {
            return UndUserDetails(
                    user.id,
                    user.username,
                    user.firstname,
                    user.lastname,
                    user.email,
                    user.password,
                    mapToGrantedAuthorities(user.authorities),
                    user.enabled,
                    user.lastPasswordResetDate,
                    user.clientSecret,
                    user.key
            )
        }

        private fun mapToGrantedAuthorities(authorities: List<Authority>?): List<GrantedAuthority> {
            return authorities!!.map { authority -> SimpleGrantedAuthority(authority.name!!.name) }

        }
    }
}
